pkg/cli/start.go

```diff
@@ -1426,3 +1426,30 @@ func reportReadinessExternally(ctx context.Context, cmd *cobra.Command, waitForI
 		log.Ops.Errorf(ctx, "failed to signal readiness using systemd protocol: %s", err)
 	}
 }
+
+// initCPUProfiler creates a new CPUProfiler and takes
+// a profile provided that the cpu profile interval has
+// been exceeded.
+func initCPUProfiler(ctx context.Context, dir string, st *cluster.Settings, stopper *stop.Stopper) {
+	cpuProfileInterval := envutil.EnvOrDefaultDuration("COCKROACH_CPUPROF_INTERVAL", -1)
+	if cpuProfileInterval <= 0 {
+		return
+	}
+	cpuProfiler, err := profiler.NewCPUProfiler(ctx, dir, st)
```

Comment:
I don't think this quite works, you're loading the env var but you're not using it, so the profiler is really just running every second. You want to sleep for the duration.

Also, didn't the old code take profiles unconditionally? So here we should be doing it unconditionally too (and update the comment).

I'm not sure where the original discussion thread went, but I think I recommended removing this code altogether since the cluster setting replaces this already, and only one CPU profile can be active at any given time. Was there any pushback?